src/jrd/jrd.cpp

```diff
@@ -7592,11 +7592,17 @@ static JAttachment* create_attachment(const PathName& alias_name,
 
 static void check_single_maintenance(thread_db* tdbb)
 {
-	UCHAR spare_memory[RAW_HEADER_SIZE + PAGE_ALIGNMENT];
-	UCHAR* header_page_buffer = FB_ALIGN(spare_memory, PAGE_ALIGNMENT);
+	Database* const dbb = tdbb->getDatabase();
+
+	const ULONG ioBlockSize = dbb->getIOBlockSize();
+	const ULONG headerSize = MAX(RAW_HEADER_SIZE, ioBlockSize);
+
+	HalfStaticArray<UCHAR, RAW_HEADER_SIZE + PAGE_ALIGNMENT> temp;
```

Comment:
Here you used HalfStaticArray with fixed alignments, not using the mechanism of its Align parameter and using the base `getAlignedBuffer`. That's the place I'm saying it's better to change HalfStaticArray and use it normally.

Reply:
The actual alignment could be different (bigger) than used in template, thus I don't see what changes in HalfStaticArray could make the code better or more efficient.